cache/blobs.go

```diff
@@ -53,11 +53,174 @@ func (sr *immutableRef) computeBlobChain(ctx context.Context, createIfNeeded boo
 	// refs rather than every single layer present among their ancestors.
 	filter := sr.layerSet()
 
-	return computeBlobChain(ctx, sr, createIfNeeded, comp, s, filter)
+	var err error
+	extraDescs := []ocispecs.Descriptor{}
+	if comp.Type == compression.NydusBlob {
+		// For nydus image, an additional bootstrap layer needs to be appended to manifest.
+		extraDescs, err = computeNydusBlobChain(ctx, sr, s)
```

Comment:
Is the behaviour of `computeNydusBlobChain` compatible to `computeBlobChain`? e.g. Is this aware of `*cacheRecord.kind()`?

Reply:
`computeNydusBlobChain` uses `sr.layerWalk` internally, it seems to be consistent with `computeBlobChain ` in the way to handle `kind()`?